File: codecheck-case/lua/BKCT_NullPointDereference/BKCT_NullPointDereference_01.lua
local function bad_01()
    local str
    str = nil
    -- bad
    print(#str)

    local tableVar
    tableVar = nil
    -- bad
    local keyCount = #tableVar

    local list = {"element1", "element2"}
    list[1] = nil
    -- bad
    print(list[1]:upper())

    local var
    var = nil
    -- bad
    print(var.someField)
end

local function good_01()
    local str = "Hello"
    -- good
    print(#str)

    local tableVar = {"element1", "element2"}
    -- good
    local keyCount = #tableVar

    local list = {"element1", "element2"}
    list[1] = "Not nil"
    -- good
    print(list[1]:upper())

    local var = {}
    var.someField = 1
    -- good
    print(var.someField)
end

File: codecheck-case/java/BKCT_NullPointDereference/BKCT_NullPointDereference_01.java
package java.BKCT_NullPointDereference;

public class BKCT_NullPointDereference_01 {
    void bad_01() {
        String str;
        str = null;
        // bad
        int length = str.length();

        Integer num;
        num = null;
        // bad
        int value = num.intValue();

        String[] arr = new String[2];
        arr[0] = null;
        // bad
        System.out.println(arr[0].charAt(0));

        Object obj;
        obj = null;
        // bad
        obj.toString();
    }

    void good_01() {
        String str = "Hello";
        // good
        int length = str.length();

        Integer num = 10;
        // good
        int value = num.intValue();

        String[] arr = new String[2];
        arr[0] = "test";
        // good
        System.out.println(arr[0].charAt(0));

        Object obj = new Object();
        // good
        obj.toString();
    }
}